misc/dbt-materialize/tests/adapter/test_deploy.py

```diff
@@ -0,0 +1,316 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License in the LICENSE file at the
+# root of this repository, or online at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+from dbt.tests.util import run_dbt
+from fixtures import (
+    test_materialized_view,
+    test_materialized_view_index,
+    test_sink,
+    test_source,
+    test_view_index,
+)
+
+
+class TestRunWithDeploy:
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "vars": {
+                "deployment": {
+                    "default": {"clusters": ["quickstart"], "schemas": ["public"]}
+                }
+            }
+        }
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "test_materialized_view_index.sql": test_materialized_view_index,
+            "test_view_index.sql": test_view_index,
+        }
+
+    def test_deployment_run(self, project):
+        # the test runner overrides schemas
+        # so we can only validate the cluster
+        # configuration is overriden.
```

Comment:
I did however, validate this works manually!

Reply:
Actually, I've run into an issue. When specifying generate_schema_name as a macro in the adapter, it's not picked up by downstream projects. I'm not really sure how to debug this but will take a look.

Reply:
You might need those downstream projects to adjust their project-level dispatch config to look in `dbt-materialize` for global `dbt` macros? Or something like that. The dispatch stuff confuses me greatly. The docs are here though: https://docs.getdbt.com/reference/dbt-jinja-functions/dispatch#overriding-global-macros

Reply:
Close. You need to read those docs backwards because they show how to make the adapter the default if you are a project. We want to force the adapter to be the default on our projects. I need to rename the macro `materialize__generate_schema_name`